File: Python/Sorting/bubblesort.py
def bubbleSort(nums):
    n = len(nums)
    for i in range(n):
        for j in range(n-i-1):
            if nums[j] > nums[j+1]:
                nums[j], nums[j+1] = nums[j+1], nums[j]
    return nums

nums = []
n = int(input("Enter n: "))
for i in range(n):
    element = int(input("Enter element {}: ".format(i+1)))
    nums.append(element)

print("Current array:", nums)
bubbleSort(nums)
print("Sorted array:", nums)

File: C/Sorting/bubble-sort.c
#include <stdio.h>

void bubbleSort(int arr[], int n);
void display(int arr[], int n);

int main() {

    int arr[] = {4, 7, 1, 2, 8};
    bubbleSort(arr, 5);

}

void bubbleSort(int arr[], int n) {

    int i, j; n--;
    
    for (i = 0; i < n; ++i) {
        for (j = 0; j < n-i; ++j) {
            if (arr[j] > arr[j+1]) {
                int temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
        display(arr, 5);
    }

}

void display(int arr[], int n) {

    printf("Current array: ");
    for (int i = 0; i < n; ++i) {
        printf("%d ", arr[i]);
    }
    printf("\n");

}